subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/AbsolutePathFingerprintingStrategy.java

```diff
@@ -67,7 +69,7 @@ public boolean preVisitDirectory(DirectorySnapshot directorySnapshot) {
 
                 @Override
                 public void visitFile(FileSystemLocationSnapshot fileSnapshot) {
-                    if (!includeMissing && fileSnapshot.getType() == FileType.Missing) {
+                    if (isRoot() && (!includeMissing && fileSnapshot.getType() == FileType.Missing)) {
```

Comment:
```suggestion
                    if (isRoot() && !includeMissing && fileSnapshot.getType() == FileType.Missing) {
```

I think the brackets are unnecessary.

Reply:
Just wanted to group the include missing flag with checking for missing, to make sure they're interpreted together. Would you like me to split it?

Reply:
I think `includeMissing` means now `includeMissingRoots` and I would probably move it to the first position. I don't think the brackets help much in interpreting the boolean condition.

Reply:
If you don't think it helps, I'll change it :)

Reply:
Done

Reply:
Renamed to `includeMissingRoots`